service-api/app/test/AppTest/Handler/LpasActionsHandlerTest.php

```diff
@@ -0,0 +1,116 @@
+            ->setMethods(['__invoke'])->getMock();
+
+
+        $requestData = [
+            'reference_number' => 'number',
+            'dob' => ['d','o','b'],
+            'first_names' => 'first name',
+            'last_name' => 'last name',
+            'postcode' => 'postcode'
+        ];
+
+        $matchedLPAData = [
+            'lpa-id' => 'number',
+            'actor-id' => '123',
+        ];
+
+        $handler = new LpasActionsHandler($olderLpaServiceProphecy->reveal(), $featureEnabledProphecy);
+
+        $requestProphecy = $this->prophesize(ServerRequestInterface::class);
+        $requestProphecy->getParsedBody()
+            ->willReturn($requestData);
+        $requestProphecy->getAttribute('actor-id')->willReturn('123');
+
+        $olderLpaServiceProphecy->checkLPAMatchAndGetActorDetails('123', $requestData)->willReturn($matchedLPAData);
+
+        $featureEnabledProphecy->method('__invoke')->willReturn(true);
+
+        $olderLpaServiceProphecy->storeLPARequest('number', '123', '123')->shouldBeCalled()->willReturn('recordId');
+
+        $olderLpaServiceProphecy->hasActivationCode('number', '123')->willReturn(null);
+
+        $olderLpaServiceProphecy->requestAccessByLetter('number', '123')->willThrow(new ApiException('Uh-oh'));
+
+        $exceptionRaised = false;
+        try {
+            $handler->handle($requestProphecy->reveal());
+        } catch (ApiException $apiException) {
+            $exceptionRaised = true;
+            $olderLpaServiceProphecy->removeLpaRequest('recordId')->shouldBeCalled();
+        }
+
+        if (!$exceptionRaised) {
+            self::fail('Exception should be raised when there is a bad response from sirius');
+        }
+    }
+
+    public function test_no_records_stored_or_removed_when_feature_flag_is_false()
+    {
+        $olderLpaServiceProphecy = $this->prophesize(OlderLpaService::class);
+        $featureEnabledProphecy =  $this->getMockBuilder(FeatureEnabled::class)
+            ->setConstructorArgs([['save_older_lpa_requests']])
+            ->setMethods(['__invoke'])->getMock();
+
+
+        $requestData = [
+            'reference_number' => 'number',
+            'dob' => ['d','o','b'],
+            'first_names' => 'first name',
+            'last_name' => 'last name',
+            'postcode' => 'postcode'
+        ];
+
+        $matchedLPAData = [
+            'lpa-id' => 'number',
+            'actor-id' => '123',
+        ];
+
+        $handler = new LpasActionsHandler($olderLpaServiceProphecy->reveal(), $featureEnabledProphecy);
+
+        $requestProphecy = $this->prophesize(ServerRequestInterface::class);
+        $requestProphecy->getParsedBody()
+            ->willReturn($requestData);
+        $requestProphecy->getAttribute('actor-id')->willReturn('123');
+
+        $olderLpaServiceProphecy->checkLPAMatchAndGetActorDetails('123', $requestData)->willReturn($matchedLPAData);
+
+        $featureEnabledProphecy->method('__invoke')->willReturn(false);
```

Comment:
```suggestion
        $featureEnabledProphecy->__invoke()->willReturn(false);
```
Works and is used elsewhere in the codebase.

Reply:
Oh, sorry, bad suggestion. You'll need to pass in the expected parameter to 'invoke' as well. i.e. 'save-lpa-requests' or whatever it is.